psa-crypto/src/operations/cipher.rs

```diff
@@ -2,3 +2,146 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! # Unauthenticated Ciphers operations
+
+use crate::initialized;
+use crate::types::algorithm::Cipher;
+use crate::types::key::Id;
+use crate::types::status::{Result, Status};
+
+fn crypt(
+    encrypt: bool,
+    key_id: Id,
+    alg: Cipher,
+    plaintext: &[u8],
+    iv: &[u8],
+    ciphertext: &mut [u8],
+) -> Result<usize> {
+    initialized()?;
+
+    let mut operation: psa_crypto_sys::psa_cipher_operation_t =
+        unsafe { psa_crypto_sys::psa_cipher_operation_init() };
+
+    Status::from(unsafe {
+        (if encrypt {
+            psa_crypto_sys::psa_cipher_encrypt_setup
+        } else {
+            psa_crypto_sys::psa_cipher_decrypt_setup
+        })(&mut operation, key_id.0, alg.into())
+    })
+    .to_result()?;
+
+    Status::from(unsafe {
+        psa_crypto_sys::psa_cipher_set_iv(&mut operation, iv.as_ptr(), iv.len())
+    })
+    .to_result()?;
```

Comment:
I think for every call following a successful `psa_cipher_operation_init`, if the operation fails you need to call `psa_cipher_abort` before returning.